File: snippet/query-examples/objc/har-get.m
#import <Foundation/Foundation.h>

NSDictionary *headers = @{ @"User-Agent": @"curl/7.68.0",
                           @"Accept": @"*/*" };

NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:[NSURL URLWithString:@"https://will-x86.com/api/pixelsq?x=1&y=2&hex=%23102932"]
                                                       cachePolicy:NSURLRequestUseProtocolCachePolicy
                                                   timeoutInterval:10.0];
[request setHTTPMethod:@"GET"];
[request setAllHTTPHeaderFields:headers];

NSURLSession *session = [NSURLSession sharedSession];
NSURLSessionDataTask *dataTask = [session dataTaskWithRequest:request
                                            completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
                                                if (error) {
                                                    NSLog(@"%@", error);
                                                } else {
                                                    NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *) response;
                                                    NSLog(@"%@", httpResponse);
                                                }
                                            }];
[dataTask resume];

File: snippet/query-examples/swift/har-get.swift
import Foundation

let headers = [
  "User-Agent": "curl/7.68.0",
  "Accept": "*/*"
]

let request = NSMutableURLRequest(url: NSURL(string: "https://will-x86.com/api/pixelsq?x=1&y=2&hex=%23102932")! as URL,
                                        cachePolicy: .useProtocolCachePolicy,
                                    timeoutInterval: 10.0)
request.httpMethod = "GET"
request.allHTTPHeaderFields = headers

let session = URLSession.shared
let dataTask = session.dataTask(with: request as URLRequest, completionHandler: { (data, response, error) -> Void in
  if (error != nil) {
    print(error)
  } else {
    let httpResponse = response as? HTTPURLResponse
    print(httpResponse)
  }
})

dataTask.resume()
